build system housekeeping

eqvinox opened the pull request:
- make xrelfo warnings hard failures as requested in #8412 
- ditch some unused things on `configure`
- make the CPU time stuff runtime options & try `CLOCK_THREAD_CPUTIME_ID`

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/618c98dec81b92a2ed2557a5a5881c21/raw/7cf340388014493d9c2ee266aa6f9ec8670d6b02/cr_8463_1618340553.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/command.c b/lib/command.c
index b7eafea6e..309ea9269 100644
--- a/lib/command.c
+++ b/lib/command.c
@@ -447,7 +447,7 @@ static int config_write_host(struct vty *vty)
 		if (!cputime_threshold)
 			vty_out(vty, "no service cputime-warning\n");
 #if defined(CONSUMED_TIME_CHECK) && CONSUMED_TIME_CHECK != 5000000
-		else	/* again, always print non-default */
+		else /* again, always print non-default */
 #else
 		else if (cputime_threshold != 5000000)
 #endif
diff --git a/lib/thread.c b/lib/thread.c
index a7075c5e6..1ad927924 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -194,7 +194,8 @@ static void cpu_record_print(struct vty *vty, uint8_t filter)
 	struct listnode *ln;
 
 	if (!cputime_enabled)
-		vty_out(vty, "\n"
+		vty_out(vty,
+			"\n"
 			"Collecting CPU time statistics is currently disabled.  Following statistics\n"
 			"will be zero or may display data from when collection was enabled.  Use the\n"
 			"  \"service cputime-stats\"  command to start collecting data.\n");
@@ -342,24 +343,22 @@ DEFUN (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -368,12 +367,11 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {
@@ -1786,7 +1784,7 @@ unsigned long thread_consumed_time(RUSAGE_T *now, RUSAGE_T *start,
 {
 #ifdef HAVE_CLOCK_THREAD_CPUTIME_ID
 	*cputime = (now->cpu.tv_sec - start->cpu.tv_sec) * TIMER_SECOND_MICRO
-		+ (now->cpu.tv_nsec - start->cpu.tv_nsec) / 1000;
+		   + (now->cpu.tv_nsec - start->cpu.tv_nsec) / 1000;
 #else
 	/* This is 'user + sys' time.  */
 	*cputime = timeval_elapsed(now->cpu.ru_utime, start->cpu.ru_utime)
diff --git a/lib/vty.c b/lib/vty.c
index 5e0c074b2..5b72f6691 100644
--- a/lib/vty.c
+++ b/lib/vty.c
@@ -511,11 +511,12 @@ static int vty_command(struct vty *vty, char *buf)
 
 	if (cputime_enabled && cputime_threshold
 	    && (realtime = thread_consumed_time(&after, &before, &cputime))
-		    > cputime_threshold)
+		       > cputime_threshold)
 		/* Warn about CPU hog that must be fixed. */
-		flog_warn(EC_LIB_SLOW_THREAD,
-			  "SLOW COMMAND: command took %lums (cpu time %lums): %s",
-			  realtime / 1000, cputime / 1000, buf);
+		flog_warn(
+			EC_LIB_SLOW_THREAD,
+			"SLOW COMMAND: command took %lums (cpu time %lums): %s",
+			realtime / 1000, cputime / 1000, buf);
 
 	/* Get the name of the protocol if any */
 	protocolname = frr_protoname;

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/af0476d87e3ee0c1088a773f98ea9bfc/raw/25e8091d9f29c7e92eadd999de962674b5053376/cr_8463_1618340662.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index 912525bdd..1ad927924 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -343,24 +343,22 @@ DEFUN (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -369,12 +367,11 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {
@@ -1787,7 +1784,7 @@ unsigned long thread_consumed_time(RUSAGE_T *now, RUSAGE_T *start,
 {
 #ifdef HAVE_CLOCK_THREAD_CPUTIME_ID
 	*cputime = (now->cpu.tv_sec - start->cpu.tv_sec) * TIMER_SECOND_MICRO
-		+ (now->cpu.tv_nsec - start->cpu.tv_nsec) / 1000;
+		   + (now->cpu.tv_nsec - start->cpu.tv_nsec) / 1000;
 #else
 	/* This is 'user + sys' time.  */
 	*cputime = timeval_elapsed(now->cpu.ru_utime, start->cpu.ru_utime)

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/0ff68d09a6c5284703c4e1ca889aa352/raw/190f822d3d250a75f1945c7791a5435dcf5ee0cc/cr_8463_1618340764.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index fa3216dab..1ad927924 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -343,24 +343,22 @@ DEFUN (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -369,12 +367,11 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

donaldsharp commented on configure.ac:
```diff
@@ -2198,6 +2198,10 @@ AC_CHECK_DECL([CLOCK_MONOTONIC],
 	 AC_DEFINE([HAVE_CLOCK_MONOTONIC], [1], [Have monotonic clock])
 ], [AC_MSG_RESULT([no])], [FRR_INCLUDES])
 
+AC_CHECK_DECL([CLOCK_THREAD_CPUTIME_ID], [
```
please add this change to doc/user/installation.rst

eqvinox replied:
Uh ... it's not an option to switch?  ... it does the same thing as `getrusage()` ... it's used if it's available ... and has been around forever ... I'm honestly not sure what to put in the docs about this?

donaldsharp replied:
ignore me my brain is mush lately

donaldsharp reviewed (changes requested):
we need a doc update

qlyoung commented:
@eqvinox don't we need a doc update for the new `service xyz` options?

eqvinox commented:
> @eqvinox don't we need a doc update for the new `service xyz` options?

Oh yeah that makes more sense :laughing:

I'll stick it on the rebase, need to fix the conflicts too...

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/7bd6a6a83ff41316d020594ba82fa75d/raw/2f76452763d0b03614531552f294c06ed4a832ad/cr_8463_1618519829.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index ab9867adc..d407e08e6 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

eqvinox commented:
- rebased
- split the setting for wallclock time and cpu time warnings, can now be tuned and enabled/disabled separately (to go along with PR #8003)
- added documentation for the new knobs

eqvinox commented:
NB: the 3 added `service XYZ` commands are affected by the same problem that is fixed in #8465 - i.e. they don't currently work in vtysh (since `lib/thread.c` is not in `vtysh_scan =`).  I don't want to trip over my own work there so I'm leaving it unfixed here.  (Also the fix is completely orthogonal anyway.)

... and also I still need to figure out what I broke in #8465.

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/c9af82b873cd177a2ae32c853d592d15/raw/8cd258acf1191e5a838bbc73019acc1bdbaf2183/cr_8463_1618728814.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index 5fbbd53db..860bbbfa4 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN_NOSH (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

eqvinox commented:
@donaldsharp you still have an :X: on this, can you check and/or remove?

eqvinox commented:
@donaldsharp roses are red, violets are blue, this PR is bored, an :x: left have you

eqvinox commented:
@qlyoung 'ye feel like pushing merge?  Already went & poked Donald to dismiss his stale review ;)

donaldsharp commented on configure.ac:
```diff
@@ -2659,6 +2658,16 @@ vici socket path        : ${vici_socket}
 The above user and group must have read/write access to the state file
 directory and to the config files in the config file directory."
 
+if test -n "$enable_datacenter"; then
```
can we update installation.rst to note that these three configure options are deprecated now?

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/57a939fcf779c540b4ff0c827f20a8cc/raw/88648c9666d24685a64b3ab403223cf8d5e104e1/cr_8463_1624453250.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index adc7a432b..f4c81cdf1 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN_NOSH (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/12e2899dbe5c2ec7b3b3e80659b4df73/raw/88648c9666d24685a64b3ab403223cf8d5e104e1/cr_8463_1624453924.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index adc7a432b..f4c81cdf1 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN_NOSH (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/ac5583c80a9ddeb18109fd6736ea58c9/raw/f13bc5dea28a4793ac6e8b11dd8fd469f3087eae/cr_8463_1624521745.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index 835aa3811..0812fbe56 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN_NOSH (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

eqvinox commented:
fixed SA warning

polychaeta reviewed (commented):
Thanks for your contribution to FRR!


<details>
<summary><b>Click for style suggestions</b></summary>

To apply these suggestions:

<p>

```
curl -s https://gist.githubusercontent.com/polychaeta/572e6edea04e27cd7a973d922eff5ef0/raw/f13bc5dea28a4793ac6e8b11dd8fd469f3087eae/cr_8463_1624545799.diff | git apply
```

</p>

<p>

```diff
diff --git a/lib/thread.c b/lib/thread.c
index 835aa3811..0812fbe56 100644
--- a/lib/thread.c
+++ b/lib/thread.c
@@ -352,24 +352,22 @@ DEFUN_NOSH (show_thread_cpu,
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_stats,
-       service_cputime_stats_cmd,
-       "[no] service cputime-stats",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Collect CPU usage statistics\n")
+DEFPY(service_cputime_stats, service_cputime_stats_cmd,
+      "[no] service cputime-stats",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Collect CPU usage statistics\n")
 {
 	cputime_enabled = !no;
 	return CMD_SUCCESS;
 }
 
-DEFPY (service_cputime_warning,
-       service_cputime_warning_cmd,
-       "[no] service cputime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_cputime_warning, service_cputime_warning_cmd,
+      "[no] service cputime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		cputime_threshold = 0;
@@ -378,20 +376,18 @@ DEFPY (service_cputime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_cputime_warning,
-       no_service_cputime_warning_cmd,
-       "no service cputime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding CPU usage threshold\n")
+ALIAS(service_cputime_warning, no_service_cputime_warning_cmd,
+      "no service cputime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding CPU usage threshold\n")
 
-DEFPY (service_walltime_warning,
-       service_walltime_warning_cmd,
-       "[no] service walltime-warning (1-4294967295)",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n"
-       "Warning threshold in milliseconds\n")
+DEFPY(service_walltime_warning, service_walltime_warning_cmd,
+      "[no] service walltime-warning (1-4294967295)",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n"
+      "Warning threshold in milliseconds\n")
 {
 	if (no)
 		walltime_threshold = 0;
@@ -400,12 +396,11 @@ DEFPY (service_walltime_warning,
 	return CMD_SUCCESS;
 }
 
-ALIAS (service_walltime_warning,
-       no_service_walltime_warning_cmd,
-       "no service walltime-warning",
-       NO_STR
-       "Set up miscellaneous service\n"
-       "Warn for tasks exceeding total wallclock threshold\n")
+ALIAS(service_walltime_warning, no_service_walltime_warning_cmd,
+      "no service walltime-warning",
+      NO_STR
+      "Set up miscellaneous service\n"
+      "Warn for tasks exceeding total wallclock threshold\n")
 
 static void show_thread_poll_helper(struct vty *vty, struct thread_master *m)
 {

```

</p>
</details>

            
If you are a new contributor to FRR, please see our [contributing guidelines](http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html#coding-practices-style).

After making changes, you do not need to create a new PR. You should perform an [amend or interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) followed by a [force push](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force).

eqvinox commented:
rebase & rerun CI

(random failure in CI, cf. #8922)
